### DISCUSSION TITLE
Adding permission-based and toggleable notifications for when banned players try to join the server, muted players try to speak, etc.

### DISCUSSION THREAD

**Discussion by MrCrazys** [Zrips/CMI#7499] [Enhancement]
It would be really cool if players with a certain permission node (such as `cmi.command.ban.notifyonjoin`) are notified when a banned player tries to join. The same could apply when muted players try to speak, except the permission node is changed to `cmi.command.mute.notifyonspeak`. This could be really helpful to see if players are constantly trying to rejoin your server. This could also perhaps be a toggle in `/cmi options`. Another option could also be the interval it shows up, for example if a banned player constantly tries to rejoin and the interval is set to 5 (5 seconds), a notification would only show up every 5 seconds, to prevent spamming staff members with notifications if they are constantly rejoining.

Additionally, it would also be really cool to send players with `cmi.command.ban.evadenotify` notifications when a player with the same IP address as a banned player joins, and it shows all their other accounts.

**Comment by mrfloris** [comment]
The right time to ask, because recently I've asked zrips to consider expanding the chatfilter option by allowing us to regex against bad usernames, so we don't see certain ppl trying to join - though totally different, it's about the post-join event I guess, so this would tie in there, and could potentially be included if that makes it to cmi

**Reply by MrCrazys** [reply]
> The right time to ask, because recently I've asked zrips to consider expanding the chatfilter option by allowing us to regex against bad usernames, so we don't see certain ppl trying to join - though totally different, it's about the post-join event I guess, so this would tie in there, and could potentially be included if that makes it to cmi

Yeah, it would just be checking for when a player joins and if they are banned, IP-banned, temp-banned or temp-IP-banned, and iterating through all online players, and finally sending a message to all players with the right permission (and who have it toggled on if it was part of `/cmi options`) (I would guess, just going from what I think here). It would certainly be cool to see the chat filter expand. I personally don't use it, but it would be nice to see it evolve.
